nodes/src/nodes/db_neo4j/neo4j_driver.py

```
@@ -256,6 +397,82 @@ def _invoke_get_data(self, input_obj: Dict[str, Any]) -> Dict[str, Any]:
+        except Exception as e:
+            return {'error': str(e)}
+
+    def _invoke_run_cypher(self, input_obj: Dict[str, Any]) -> Dict[str, Any]:
+        """Execute an arbitrary Cypher statement (read or write)."""
+        cypher = input_obj['cypher'].strip()
+        params = input_obj.get('params') or {}
+
+        try:
+            # Use write path so both reads and writes succeed.
+            rows = self._instance.IGlobal._run_write_query(cypher, params)
+            return {'rows': rows}
```

Ruff (0.15.7)

[warning] Consider moving this statement to an `else` block

(TRY300)